src/features/components/templates/signup/with-banner/form/index.tsx

```diff
@@ -15,15 +15,16 @@ const SignupPublicForm = () => {
 
     const {
         register,
-        formState: { errors, isValid },
+        formState: { errors, isValid, isSubmitting, isSubmitSuccessful },
         watch,
         clearErrors,
         setValue,
         handleSubmit,
     } = signUpForm
     const values = watch()
 
-    const isButtonDisabled = values.email === '' || !values.terms || !isValid
+    const isButtonDisabled =
+        values.email === '' || !values.terms || !isValid || isSubmitting || isSubmitSuccessful
```

Comment:
its better approach if you write the code like this `isSubmitting || isSubmitSuccessful || !values.terms || !isValid || ''`
is conditions at first and not condition followed by null value

Reply:
```suggestion
     isSubmitting || isSubmitSuccessful || !values.terms || !isValid || ''
```